Sources/Danger/Danger.swift

```diff
@@ -72,24 +76,26 @@ public func Danger() -> DangerDSL {
 private var dumpInfo: (danger: DangerRunner, path: String)?
 
 private func dumpResultsAtExit(_ runner: DangerRunner, path: String) {
-    func dump() {
-        guard let dumpInfo = dumpInfo else { return }
-        dumpInfo.danger.logger.debug("Sending results back to Danger")
-        do {
-            let encoder = JSONEncoder()
-            encoder.outputFormatting = .prettyPrinted
-            let data = try encoder.encode(dumpInfo.danger.results)
-
-            if !FileManager.default.createFile(atPath: dumpInfo.path, contents: data, attributes: nil) {
-                dumpInfo.danger.logger.logError("Could not create a temporary file for the Dangerfile DSL at: \(dumpInfo.path)")
-                exit(0)
-            }
+    dumpInfo = (runner, path)
+    dump()
+    atexit(dump)
+}
 
-        } catch {
-            dumpInfo.danger.logger.logError("Failed to generate result JSON:", error)
-            exit(1)
+func dump() {
+    guard let dumpInfo = dumpInfo else { return }
+    dumpInfo.danger.logger.debug("Sending results back to Danger")
+    do {
+        let encoder = JSONEncoder()
+        encoder.outputFormatting = .prettyPrinted
+        let data = try encoder.encode(dumpInfo.danger.results)
+
+        if !FileManager.default.createFile(atPath: dumpInfo.path, contents: data, attributes: nil) {
+            dumpInfo.danger.logger.logError("Could not create a temporary file for the Dangerfile DSL at: \(dumpInfo.path)")
```

Comment:
<!--
  0 failure: 
  1 warning:  Line should be 12...
  
  
  DangerID: danger-id-default;
  File: Sources/Danger/Danger.swift;
  Line: 93;
-->

- :warning: Line should be 120 characters or less: currently 124 characters (`line_length`)